CHANGELOG.md

```diff
@@ -2,6 +2,9 @@
 
 ## unreleased
 
+- fix: Set GCC_WARN_64_TO_32_BIT_CONVERSION to YES on project level
+- fix: Fix compiler warnings regarding unused entity and implicit casts
```

Comment:
`h`: As this is a refactoring for us we don't mention it in the ReadMe. So let's please remove it.
```suggestion
```